commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -96,11 +147,29 @@ private String parseString(NetworkResponse response) {
     }
 
     private static class QueryResponse {
-        private Query query;
-
+        private Query query = new Query();
+        private Page page;
+
+        private String printSet() {
+            if (categorySet == null || categorySet.isEmpty()) {
+                gpsCatExists = false;
+                Log.d("Cat", "gpsCatExists=" + gpsCatExists);
+                return "No collection of categories";
+            }
+            else {
+                gpsCatExists = true;
+                Log.d("Cat", "gpsCatExists=" + gpsCatExists);
+                return "CATEGORIES FOUND" + categorySet.toString();
+            }
+        }
         @Override
         public String toString() {
-            return "query=" + query.toString();
+            if (query!=null) {
+                return "query=" + query.toString() + "\n" + printSet();
+            }
+            else {
```

Comment:
Is it a convention of this codebase to put else clauses on their own lines? They normally appear on the same line as the if closing bracket in Java:

```
if (...) {
    ...
} else {
   ...
}
```